CodeSignESRP.targets

```diff
@@ -0,0 +1,144 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+	<UsingTask TaskName="ESRPSignTask" AssemblyFile="$(CISignRepoPath)\tools\sdkbuildtools\tasks\MS.Az.Sdk.OnPremise.Build.Tasks.dll" />
+
+	<PropertyGroup>
+		<!-- CISignRepo is an environment variable that points to ci-signing repo clone -->
+		<CISignRepoPath>$(CISignRepo)</CISignRepoPath>
+	</PropertyGroup>
+   
+	<Target Name="CodeSignBinariesESRP" DependsOnTargets="RestoreNugetPackages;BuildMsBuildTask">
+
+    <Message Text="====> Executing CodeSignBinariesESRP Target..." Importance="high"/>
+	
+    <PropertyGroup>
+		<!--public token associated with MSSharedLibKey.snk-->
+		<StrongNameToken Condition=" '$(StrongNameToken)' == '' ">31bf3856ad364e35</StrongNameToken>
+    </PropertyGroup>
+	
+	<Message Text="----> Dlls signing section" Importance="high"/>
+
+    <!-- Azure -->
+    <ItemGroup Condition="'$(Scope)' == 'All' or '$(Scope)' == 'Latest' or '$(Scope)' == 'Netcore'">
+		<DelaySignedAssembliesToSign Include="$(PackageDirectory)\$(Configuration)\**\Microsoft*Azure*Commands*.dll" Exclude="$(PackageDirectory)\$(Configuration)\**\Microsoft*Azure*Commands*Common*.dll" />
+		<DelaySignedAssembliesToSign Include="$(PackageDirectory)\$(Configuration)\**\Microsoft.Azure.Common.Extensions.dll" />
+		<DelaySignedAssembliesToSign Include="$(PackageDirectory)\$(Configuration)\**\Microsoft.Azure.Management.Sql.Legacy.dll" />
+    </ItemGroup>
+
+    <!-- Stack -->
+    <ItemGroup Condition="'$(Scope)' == 'All' or '$(Scope)' == 'Stack'">
+		<DelaySignedAssembliesToSign Include="$(StackPackageFolder)\$(Configuration)\**\Microsoft*Azure*Commands*.dll" Exclude="$(StackPackageFolder)\$(Configuration)\**\Microsoft*Azure*Commands*Common*.dll" />
+		<DelaySignedAssembliesToSign Include="$(StackPackageFolder)\$(Configuration)\**\Microsoft.Azure.Management.Storage.dll" />
+		<DelaySignedAssembliesToSign Include="$(StackPackageFolder)\$(Configuration)\**\Microsoft.Azure.Common.Extensions.dll" />
+		<DelaySignedAssembliesToSign Include="$(StackPackageFolder)\$(Configuration)\**\Microsoft.Azure.Management.Sql.Legacy.dll" />
+    </ItemGroup>
+
+    <Message Importance="high" Text="$(PackageDirectory)\$(Configuration) does not contains any files to sign. Code sign will skip."
+        Condition="'@(DelaySignedAssembliesToSign)' == ''" />
+    
+    <ESRPSignTask
```

Comment:
How do we control which certifciates are used for strong name and authenticode signing?  Are these configurable in task parameters?